### PR TITLE
Fix package override resolver desktop

### LINKED ISSUE #1854: ResolvePackageFileConflicts throwing when referencing System.Diagnostics.DiagnosticSource by assembly path
This is new failure that just started happening in some Razor tests for our MSBuild functionality. This started failing for us `2.2.0-preview1-007927` - this was working before in `2.2.0-preview1-007866`.

I've attached a repro project, which is doing some esoteric things because this is part of our tests for our MSBuild targets 😆. I think the probably might not be anything wierd that we're doing though.

The project file looks like this:
```
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup Condition="'$(BinariesRoot)'==''">
    <!-- In test scenarios $(BinariesRoot) is defined in a generated Directory.Build.props file -->
    <ProjectReference Include="..\..\Microsoft.AspNetCore.Razor.Test.MvcShim\Microsoft.AspNetCore.Razor.Test.MvcShim.csproj"/>
    <ProjectReference Include="..\..\..\src\Microsoft.AspNetCore.Razor.Runtime\Microsoft.AspNetCore.Razor.Runtime.csproj"/>
  </ItemGroup>

  <ItemGroup Condition="'$(BinariesRoot)'!=''">
    <Reference Include="$(BinariesRoot)\System.Diagnostics.DiagnosticSource.dll"/>
    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Html.Abstractions.dll"/>
    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Razor.dll"/>
    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Razor.Runtime.dll"/>
    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Razor.Test.MvcShim.dll"/>
  </ItemGroup>
</Project>
```
Removing `<Reference Include="$(BinariesRoot)\System.Diagnostics.DiagnosticSource.dll"/>` seems to fix it.

```
Microsoft (R) Build Engine version 15.6.22.57775 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj...
  Generating MSBuild file C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\obj\SimpleMvc.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\obj\SimpleMvc.csproj.nuget.g.targets.
  Restore completed in 273.82 ms for C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj.
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018: The "ResolvePackageFileConflicts" task failed unexpectedly. [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018: System.ArgumentNullException: Value cannot be null. [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018: Parameter name: key [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.FindEntry(TKey key) [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value) [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ConflictResolution.PackageOverrideResolver`1.Resolve(TConflictItem item1, TConflictItem item2) [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ConflictResolution.ConflictResolver`1.ResolveConflict(TConflictItem item1, TConflictItem item2) [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ConflictResolution.ConflictResolver`1.ResolveConflicts(IEnumerable`1 conflictItems, Func`2 getItemKey, Action`1 foundConflict, Boolean commitWinner, Action`1 unresolvedConflict) [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ConflictResolution.ResolvePackageFileConflicts.ExecuteCore()[C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() in E:\A\_work\24\s\src\Build\BackEnd\TaskExecutionHost\TaskExecutionHost.cs:line 631 [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext() in E:\A\_work\24\s\src\Build\BackEnd\Components\RequestBuilder\TaskBuilder.cs:line 787 [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
```

```
.NET Command Line Tools (2.2.0-preview1-007927)

Product Information:
 Version:            2.2.0-preview1-007927
 Commit SHA-1 hash:  a212d5a017

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\

Microsoft .NET Core Shared Framework Host

  Version  : 2.1.0-preview1-26110-02
  Build    : 77152705cd56e7d89f52585959ec3559d91084c4
```
[Repro.zip](https://github.com/dotnet/sdk/files/1627368/Repro.zip)

### PR DESCRIPTION
**Pull request by ericstj**
In desktop projects the PackageOverrideResolver was throwing ArgumentNullException due to 
one item coming from a package and the other not.

Moreover I noticed that the one that was thought to come from a package was actually a platform item
and the reason we thought it came from a package was that I had a *.nuspec in the root of the drive where I was building from.

We need tests to cover these cases.

Fixes #1854

### CODE REVIEW

**Comment by ericstj** [comment]
test OSX10.12 Debug

File: src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAPackageOverrideResolver.cs [hunk 1]
```diff
@@ -48,5 +48,48 @@ public void ItHandlesNullITaskItemArray()
...
+
+            var packageItem2 = new MockConflictItem("System.Eric")
+            {
+                PackageId = "FakePlatform",
+                PackageVersion = new Version(4, 0, 0),
+                AssemblyVersion = new Version(4, 0, 0, 0),
+                ItemType = ConflictItemType.Reference
+            };
+
+            Assert.Null(resolver.Resolve(packageItem2, platformItem));
```
Comment (eerhardt) [inline src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAPackageOverrideResolver.cs]: Did you want to switch these around so the `platformItem` with a `null` PackageId comes first?

**Review by eerhardt — APPROVED** [review]
Thanks for fixing this.

File: src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAPackageOverrideResolver.cs (hunk 1 above)
Comment (ericstj) [reply src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAPackageOverrideResolver.cs]: No the distinction of the second test is that one item is in the override package list. I could add another call that swaps them just to ensure coverage.

**Comment by ericstj** [comment]
test OSX10.12 Release

**Comment by ericstj** [comment]
@livarcocc should this be merged to a release branch?

**Comment by livarcocc** [comment]
We will do a mop up merge of master into release/2.1.3xx.

**Comment by ericstj** [comment]
Thanks

**Pull request merged**